Negative river discharge

**Describe the bug**

When I ran ve_example to a longer time length, at the 50th month it errored
```
ValueError: The river discharge should not be negative!
```
which stemmed from
```
    subsurface_runoff_routed_plus_local = above_ground.route_horizontal_flow(
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Any idea the river is running dry? (or soaking up water?)

**To Reproduce**

Run VE for longer by:
1. Install ve_example
2. add longer time length to config
```
[core.timing]
run_length = "10 years"
```
3. make sure the example data have matching time dimensions by changing `time` in `common.py` to
```
time = np.arange(np.datetime64("2013-01"), np.datetime64("2023-01")).astype(
    "datetime64[D]"
)
```
4. run ve_run

I ran it twice and both time it crashed at the 50th month step.

**Desktop (please complete the following information):**

- OS: Windows 10 x64
- VE dev version

Hi, bumping this again because the same error resurfaced when we run the Maliau data subset (10x10 grids) using the latest release [v0.1.1a17](https://github.com/ImperialCollegeLondon/virtual_ecosystem/releases/tag/v0.1.1a17_2026_04_16)

At **time step 71/132** (sometimes til 76/132), the log ended with VE crashing:
```
[ERROR] - above_ground - route_horizontal_flow(405) - The river discharge should not be negative!
```

The terminal message suggests the same source of error when assigning 
```
subsurface_runoff_routed_plus_local = above_ground.route_horizontal_flow()
```

@vgro any idea?

Reproducing the error using VE [v0.1.1a17] @hrlai @lelavathy 
VE version: v0.1.1a17_2026_04_16

The model fails during the hydrology update step:  
`[INFO] - hydrology_model - update(...) - Updating hydrology model`

followed by  
`[ERROR] - above_ground - route_horizontal_flow(...) - The river discharge should not be negative!`

### Logfile (maliau_2, 10x10 grid)

<details>
  <summary>Click to expand full logfile</summary>

```text
[INFO] - config_builder - Config paths resolve to 5 files
[INFO] - config_builder - Config TOML loaded from animal_config.toml
[INFO] - config_builder - Config TOML loaded from data_config.toml
[INFO] - config_builder - Config TOML loaded from plant_config.toml
[INFO] - config_builder - Config TOML loaded from soil_microbial_groups.toml
[INFO] - config_builder - Config TOML loaded from ve_run.toml

[INFO] - main - All models successfully initialised.

[INFO] - variables - Model update execution order set:
plants → abiotic_simple → animal → hydrology → litter → soil

[INFO] - main - Starting update 0: 2010-01-01

[INFO] - hydrology_model - surface_runoff_routed_plus_local added
[INFO] - hydrology_model - subsurface_runoff_routed_plus_local added
[INFO] - hydrology_model - total_runoff added

[WARNING] - hydrology_model - Update interval is not a whole number of days (30.4375)

[INFO] - main - Starting update 1: 2010-01-31
[INFO] - main - Starting update 2: 2010-03-02
[INFO] - main - Starting update 3: 2010-04-02

...

> Hi, bumping this again because the same error resurfaced when we run the Maliau data subset (10x10 grids) using the latest release [v0.1.1a17](https://github.com/ImperialCollegeLondon/virtual_ecosystem/releases/tag/v0.1.1a17_2026_04_16)
> 
> At **time step 71/132** (sometimes til 76/132), the log ended with VE crashing:
> 
> ```
> [ERROR] - above_ground - route_horizontal_flow(405) - The river discharge should not be negative!
> ```
> 
> The terminal message suggests the same source of error when assigning
> 
> ```
> subsurface_runoff_routed_plus_local = above_ground.route_horizontal_flow()
> ```
> 
> [@vgro](https://github.com/vgro) any idea?

Have you had a look at the actual values? I think this could just be a rounding error, but if it gets very negative I can set a constraint

--- a/tests/models/hydrology/test_below_ground.py
+++ b/tests/models/hydrology/test_below_ground.py
@@ -129,3 +129,34 @@ def test_update_groundwater_storage(dummy_climate_data, fixture_hydrology_consta
     np.testing.assert_allclose(result["groundwater_storage"], exp_groundwat, rtol=1e-05)
     np.testing.assert_allclose(result["subsurface_flow"], exp_upper_flow, rtol=1e-05)
     np.testing.assert_allclose(result["baseflow"], exp_lower_flow, rtol=1e-5)
+
+
+def test_upper_flow_clamped_to_zero(fixture_hydrology_constants):
+    """Ensure negative groundwater outflow is clamped to zero."""
+
+    from virtual_ecosystem.models.hydrology.below_ground import (
+        update_groundwater_storage,
+    )
+
+    # Force a negative scenario
+    groundwater_storage = np.array(
+        [
+            [-10, -5, -1, -0.1],  # upper zone goes negative
+            [100, 100, 100, 100],  # lower zone normal
+        ]
+    )
+
+    result = update_groundwater_storage(
+        groundwater_storage=groundwater_storage,
+        vertical_flow_to_groundwater=np.zeros(4),
+        bypass_flow=np.zeros(4),
+        max_percolation_rate_uzlz=fixture_hydrology_constants.max_percolation_rate_uzlz,
+        groundwater_loss=fixture_hydrology_constants.groundwater_loss,
+        reservoir_const_upper_groundwater=fixture_hydrology_constants.reservoir_const_upper_groundwater,
+        reservoir_const_lower_groundwater=fixture_hydrology_constants.reservoir_const_lower_groundwater,
+    )
+
+    # This is the key assertion:
+    assert np.all(result["subsurface_flow"] >= 0)
+    assert np.all(result["baseflow"] >= 0)
+    assert np.all(result["groundwater_storage"][0] >= 0)

--- a/virtual_ecosystem/models/hydrology/below_ground.py
+++ b/virtual_ecosystem/models/hydrology/below_ground.py
@@ -357,15 +357,17 @@ def update_groundwater_storage(
     )
 
     # Calculate outflow from the upper zone to the channel, [mm]
-    output["subsurface_flow"] = upper_zone / reservoir_const_upper_groundwater
+    output["subsurface_flow"] = np.maximum(
+        0.0, upper_zone / reservoir_const_upper_groundwater
+    )
 
-    # Update water stored in lower zone, [mm]
+    # Update water stored in lower zone, [mm], must be >=0
     lower_zone = np.array(
         groundwater_storage[1] + percolation_to_lower_zone - groundwater_loss
     )
 
-    # Calculate outflow from the lower zone to the channel, [mm]
-    output["baseflow"] = lower_zone / reservoir_const_lower_groundwater
+    # Calculate outflow from the lower zone to the channel, [mm], must be >=0
+    output["baseflow"] = np.maximum(0.0, lower_zone / reservoir_const_lower_groundwater)
 
     # Update ground water storage
     output["groundwater_storage"] = np.vstack((upper_zone, lower_zone))

